plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -2115,7 +2115,7 @@ fc::variant read_only::get_block_header_state(const get_block_header_state_param
 
 void read_write::push_block(read_write::push_block_params&& params, next_function<read_write::push_block_results> next) {
    try {
-      auto b = std::make_shared<signed_block>( std::move(params) );
+      auto b = std::make_shared<signed_block>( params.clone() );
```

Comment:
Here you are receiving a rvalue, so there is no reason to use `clone()` I think, and I feel that `std::move(params)` is appropriate.

Reply:
There is no reason in this case and `std::move` is perfectly fine. However, in the future it would be nice to `delete` the move constructor and move operator=. This doing an explicit copy gets us one step closer. Performance here is no issue, as this has never been used. Who knows if it actually even works. We no tests for it.